packages/sanity/src/core/studio/components/navbar/search/components/common/SearchWrapper.tsx

```diff
@@ -44,7 +44,7 @@ export function SearchWrapper({
    * Set shared `onClose` in search context
    */
   useEffect(() => {
-    setOnClose(handleClose)
+    setOnClose(() => handleClose)
```

Comment:
this doesn't seem right: surely `handleClose` should be called, no?

Reply:
No, it used to be set as a ref, now we use useState. When using useState you have to use the function callback version of setState otherwise it'll call your function and set the state to whatever your function return.

The reason we use state instead of refs here is because we want the memoized context value to change when `setOnClose` is called 🙌

Reply:
Ah, thanks, that makes sense (migth be worth adding that as a comment since it stands out as something that might not have been intentional)